python/foglamp/services/core/api/backup_restore.py

```diff
@@ -7,113 +7,148 @@
 from unittest.mock import MagicMock
-Backup = MagicMock()
+Backup_Mock = MagicMock()
+
 
 __author__ = "Vaibhav Singhal"
 __copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
 __license__ = "Apache 2.0"
 __version__ = "${VERSION}"
 
+__DEFAULT_LIMIT = 20
+__DEFAULT_OFFSET = 0
 
 _help = """
     -----------------------------------------------------------------------------------
     | GET, POST       | /foglamp/backup                                                |
-    | GET             | /foglamp/backup/{backup-id}                                    |
-    | DELETE          | /foglamp/backup/{backup-id}                                    |
-    |                                                                                  |
+    | GET, DELETE     | /foglamp/backup/{backup-id}                                    |
     | PUT             | /foglamp/backup/{backup-id}/restore                            |
     -----------------------------------------------------------------------------------
 """
 
+
+class Status(IntEnum):
+    """Enumeration for backup.status"""
+    RUNNING = 1
+    COMPLETED = 2
+    CANCELLED = 3
+    INTERRUPTED = 4
+    FAILED = 5
+    RESTORED = 6
+
+
 async def get_backups(request):
-    """
-    Returns a list of all backups
+    """ Returns a list of all backups
 
-    :Example: curl -X GET  http://localhost:8082/foglamp/backup
-    :Example: curl -X GET  http://localhost:8082/foglamp/backup?limit=2&skip=1&status=complete
+    :Example: curl -X GET http://localhost:8081/foglamp/backup
+    :Example: curl -X GET http://localhost:8081/foglamp/backup?limit=2&skip=1&status=completed
     """
     try:
-        limit = int(request.query['limit']) if 'limit' in request.query else None
-        skip = int(request.query['skip']) if 'skip' in request.query else None
-        status = request.query['status'] if 'status' in request.query else None
-        # TODO : Fix after actual implementation
-        Backup.get_backup_list.return_value = [{'id': 28, 'date': '2017-08-30 04:05:10.382', 'status': 'running'},
-                                               {'id': 27, 'date': '2017-08-29 04:05:13.392', 'status': 'failed'},
-                                               {'id': 26, 'date': '2017-08-28 04:05:08.201', 'status': 'complete'}]
-
-        # backup_json = [{"id": b[0], "date": b[1], "status": b[2]}
-        #                for b in Backup.get_backup_list(limit=limit, skip=skip, status=status)]
-        backup_json = Backup.get_backup_list(limit=limit, skip=skip, status=status)
-    except Backup.DoesNotExist:
-        raise web.HTTPNotFound(reason='No backups found for queried parameters')
-    return web.json_response({"backups": backup_json})
+        backup = Backup(connect.get_storage())
+        limit = int(request.query['limit']) if 'limit' in request.query else __DEFAULT_LIMIT
+        skip = int(request.query['skip']) if 'skip' in request.query else __DEFAULT_OFFSET
```

Comment:
What if limit and offset with bad data?

Query: `http://localhost:8081/foglamp/backup?limit=blah`
Right now it throws ValueError Exception (`"message": "invalid literal for int() with base 10: 'blah'",`), but I think we should raise like below to make more appropriate message.

```
 if 'limit' in request.query:
        if request.query.get('limit').isdigit():
            limit = int(request.query.get('limit'))
 else:
       raise web.HTTPBadRequest(reason="limit must be an integer")
```